js_modules/dagster-ui/packages/ui-core/src/runs/CapturedLogPanel.tsx

```diff
@@ -33,28 +35,85 @@ export const CapturedOrExternalLogPanel = React.memo(
       (props.visibleIOType === 'stdout'
         ? logCaptureInfo.externalStdoutUrl
         : logCaptureInfo.externalStderrUrl);
-    if (externalUrl) {
+    const uriOrPath =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutUriOrPath
+        : logCaptureInfo.stderrUriOrPath);
+    const shellCmd =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutShellCmd
+        : logCaptureInfo.stderrShellCmd);
+
+    const copy = useCopyToClipboard();
+    const onClickFn = async (key: string, value: string | undefined) => {
+      if (!value) {
+        return;
+      }
+      copy(value);
+      await showSharedToaster({
+        intent: 'success',
+        icon: 'done',
+        message: `${key} string copied!`,
+      });
+    };
+    const onClickExternalUri = async () => onClickFn('log artefact URI', uriOrPath);
+    const onClickShellCmd = async () => onClickFn('shell command', shellCmd);
+
+    if (externalUrl || uriOrPath || shellCmd) {
       return (
         <Box
-          flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 1}}
+          flex={{direction: 'column', alignItems: 'center', justifyContent: 'center', gap: 1}}
           background={Colors.tooltipBackground()}
           style={{color: Colors.tooltipText(), flex: 1, minHeight: 0}}
         >
-          View logs at
-          <a
-            href={externalUrl}
-            target="_blank"
-            rel="noreferrer"
-            style={{
-              color: Colors.tooltipText(),
-              textDecoration: 'underline',
-              marginLeft: 4,
-              marginRight: 4,
-            }}
-          >
-            {externalUrl}
-          </a>
-          <Icon name="open_in_new" color={Colors.tooltipText()} size={20} style={{marginTop: 2}} />
+          {externalUrl ? (
```

Comment:
this will definitely need css+html rework, i was just passing information and making the click to copy work.

when thinking about this ui, should we provide even more information, such as additional text about to use the shell command (ie, requiring a preconfigured azure cli, for example) ? I stopped short of this because I was already concerned about increasing the event size, but maybe I could add a 'log type' field that the UI could use to render that additional information without producing it from the ComputeLogManager

Reply:
I think it's reasonable to have a log_type field or something like that, that the UI could use rather than put that information in the event body itself.